lunatrace/bsl/ingest-worker/pkg/vulnerability/processor.go

```diff
@@ -0,0 +1,283 @@
+package vulnerability
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"hash/fnv"
+	"os"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/go-jet/jet/v2/postgres"
+	"github.com/go-shiori/go-readability"
+	"github.com/rs/zerolog/log"
+	"github.com/schollz/progressbar/v3"
+	"go.uber.org/fx"
+
+	"github.com/lunasec-io/lunasec/lunatrace/gogen/sqlgen/lunatrace/vulnerability/model"
+	"github.com/lunasec-io/lunasec/lunatrace/gogen/sqlgen/lunatrace/vulnerability/table"
+)
+
+type ProcessedReference struct {
+	VulnerabilityID string `json:"id"`
+	URL             string `json:"url"`
+	Title           string `json:"title"`
+	Content         string `json:"content"`
+}
+
+type ProcessedVulnerability struct {
+	ID         string               `json:"id"`
+	Details    string               `json:"details"`
+	Summary    string               `json:"summary"`
+	References []ProcessedReference `json:"references"`
+}
+
+type VulnerabilityInfo struct {
+	model.Vulnerability
+	References []model.Reference `json:"references"`
+}
+
+type Processor interface {
+	ProcessVulnerabilityID(vulnID string) (*ProcessedVulnerability, error)
+	ProcessAllVulnerabilities(outDir string) error
+}
+
+type processorDeps struct {
+	fx.In
+	DB *sql.DB
+}
+
+type processor struct {
+	deps processorDeps
+}
+
+func hashFilename(s string) uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(s))
+	return h.Sum32()
+}
+
+func (p *processor) processVulnerability(vulnInfo VulnerabilityInfo) (*ProcessedVulnerability, error) {
```

Comment:
Similar to my comment in the other PR, but it isn't clear to anyone except insiders that "process" is different from "ingest". It certainly doesnt provide any useful context.  I would call this "scrapeAdvisories" or something similar. 

Applies to the file name as well as the function names here